runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/resources/PipelineResources.java

```diff
@@ -64,6 +70,31 @@
     return resourcePath -> !resourcePath.contains("gradle/wrapper");
   }
 
+  /**
+   * Goes through the list of files that need to be staged on runner. Removes nonexistent
+   * directories and packages existing ones. This is necessary for runners that require filesToStage
+   * to be jars only.
+   *
+   * @param options options object with the files to stage and temp location for staging
```

Comment:
We should mention how this method can mutate `options`. Or we could instead return a list of prepared files as before so it's more explicit, like `options.setFilesToStage(PipelineResources.prepareFilesForStaging(options))`.

Reply:
good idea, done